CMS-Enterprise/sbom-harbor
# feat(add-rust-support): Added Rust support

## Summary

![image](https://github.com/CMS-Enterprise/sbom-harbor/assets/6561793/3cf8ce7a-8ed9-4f32-b7d6-11154ae76ae1) This PR should go in _after_ `feat(purl-generation): Syft Service now creates a well formed, sensible purl for the Package #124` 

Added support for projects containing a Cargo.lock file

### Added

- The ability for the GitHub SBOM Ingest Provider to produce SBOMs when a Cargo.lock file is found
- New GitHub repo in the harbor-test-org.  This Repo has a Cargo.lock file to allow SBOM creation with the 
  Rust cataloger: `https://github.com/harbor-test-org/rust-repo-with-cargo-lock` 

### Changed

.gitignore no longer ignores the Cago.lock file and the Cargo.lock file is checked in.  At this time the Cargo.lock file is about ~86k and the cargo book supports checking in the lock file if the project generates an app rather than a library.

```
If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore. If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git.
```
https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

## How to test

1. Initialize test environment
2. Run `cargo test commands::ingest::github::test::execute_on_harbor_test_org`
3. Open Compass, select the Package collection and apply this filter: `{ kind: "primary", purl: /cargo/ }`
4. Ensure a single document with the purl: `pkg:cargo/harbor-test-org/rust-repo-with-cargo-lock@d0.0.0` exists
